Upgrade  to Typesettable@3.0.2

themadcreator opened the pull request:
The new typesettable includes performance updates (removed superfluous clippaths) as well as built-in support for canvas.

Since the usage is still all SVG, all plots should be identical.

themadcreator committed:
Switch to typesettable library

themadcreator committed:
Fix cast

themadcreator committed:
Fix legend tests

hellochar reviewed (approved):
Looks pretty straightforward!

Would it be difficult to add an HTML context to typesettable to typeset <span> elements or the like?

themadcreator commented:
Since we do manual positioning of the text SVG elements, we probably have all we need to do HTML rendering. I'll give it a shot.

themadcreator committed:
Fix label tests for Firefox+Linux

Firefox on linux has additional kerning between
characters that is not accounted for by
typesetter's measurer. So we just add a fudge
factor of 1px per character for that test

themadcreator committed:
merge
